knimin/lib/data_access.py

```diff
@@ -3191,6 +3186,252 @@ def delete_sample_plate(self, id):
+                 ('311318B'), ('109375A'), ('3076189')"""
+        self._con.execute(sql)
+        sql = """INSERT INTO pm.tm50_8_tool (name) VALUES ('108364Z'),
+                 ('311426B'), ('311441B'), ('409172Z')"""
+        self._con.execute(sql)
+        sql = """INSERT INTO pm.extraction_kit_lot (name) VALUES ('PM16B11')"""
+        self._con.execute(sql)
+        sql = """INSERT INTO pm.master_mix_lot (name) VALUES ('14459')"""
+        self._con.execute(sql)
+        sql = """INSERT INTO pm.water_lot (name) VALUES ('RNBD9959')"""
+        self._con.execute(sql)
+        # barcode => sample
+        sql = """INSERT INTO pm.sample (sample_id)
+                 SELECT barcode
+                 FROM barcodes.barcode;"""
+        self._con.execute(sql)
+        sql = """UPDATE pm.sample
+                 SET barcode = sample_id"""
+        self._con.execute(sql)
+        # project => study
+        sql = """INSERT INTO pm.study (qiita_study_id, title)
+                 SELECT project_id, project
+                 FROM barcodes.project;"""
+        self._con.execute(sql)
+        # project_barcode => study_sample
+        sql = """INSERT INTO pm.study_sample (study_id, sample_id)
+                 SELECT study_id, barcode
+                 FROM barcodes.project_barcode
+                 JOIN pm.study ON (project_id = qiita_study_id)
+                 ORDER BY study_id, barcode;"""
+        self._con.execute(sql)
+        # plate => plate
+        sql = """SELECT plate_id, plate
+                 FROM barcodes.plate
+                 ORDER BY plate_id"""
+        plates = [(x[0], x[1]) for x in self._con.execute_fetchall(sql)]
+        if not plates:
+            return 1
+        for plate in plates:
+            sql = """INSERT INTO pm.sample_plate (name, email, plate_type_id)
+                     VALUES (%s, 'test', 1)
+                     RETURNING sample_plate_id"""
+            id = self._con.execute_fetchone(sql, [plate[1]])[0]
+            sql = """SELECT barcode
+                     FROM barcodes.plate_barcode
+                     WHERE plate_id = %s
+                     ORDER BY barcode"""
+            barcodes = [x[0] for x in self._con.execute_fetchall(sql, [id])]
+            if not barcodes:
+                continue
+            count = 0
+            nbarcode = len(barcodes)
+            (ncol, nrow) = (12, 8)
+            sql = """INSERT INTO pm.sample_plate_layout (sample_plate_id,
+                                                         sample_id, col, row)
+                     VALUES (%s, %s, %s, %s)"""
+            for i in range(ncol):
+                for j in range(nrow):
+                    self._con.execute(sql, [id, barcodes[count], i+1, j+1])
+                    count += 1
+                    if count == nbarcode:
+                        break
+                if count == nbarcode:
+                    break
+
+    def set_deposited_ebi(self):
```

Comment:
I didn't write this function. It was there in the old labadmin. For some version control issue it re-appeared (I am yet to be more familiar with GitHub...).